packages/openapi-v3/src/parameter-decorator.ts

```diff
@@ -0,0 +1,408 @@
+     */
+    integer: createParamShortcut('path', builtinTypes.integer),
+    /**
+     * Define a parameter of "long" type that's read from request path.
+     * Usage: `@param.path.long('paramName')`
+     *
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    long: createParamShortcut('path', builtinTypes.long),
+    /**
+     * Define a parameter of "float" type that's read from request path.
+     * Usage: `@param.path.float('paramName')`
+     *
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    float: createParamShortcut('path', builtinTypes.float),
+    /**
+     * Define a parameter of "double" type that's read from request path.
+     * Usage: `@param.path.double('paramName')`
+     *
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    double: createParamShortcut('path', builtinTypes.double),
+    /**
+     * Define a parameter of "byte" type that's read from request path.
+     * Usage: `@param.path.byte('paramName')`
+     *
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    byte: createParamShortcut('path', builtinTypes.byte),
+    /**
+     * Define a parameter of "binary" type that's read from request path.
+     * Usage: `@param.path.binary('paramName')`
+     *
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    binary: createParamShortcut('path', builtinTypes.binary),
+    /**
+     * Define a parameter of "date" type that's read from request path.
+     * Usage: `@param.path.date('paramName')`
+     *
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    date: createParamShortcut('path', builtinTypes.date),
+    /**
+     * Define a parameter of "dateTime" type that's read from request path.
+     * Usage: `@param.path.dateTime('paramName')`
+     *
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    dateTime: createParamShortcut('path', builtinTypes.dateTime),
+    /**
+     * Define a parameter of "password" type that's read from request path.
+     * Usage: `@param.path.password('paramName')`
+     *
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    password: createParamShortcut('path', builtinTypes.password),
+  };
+
+  /**
+   * Define a parameter of `array` type.
+   *
+   * @example
+   * ```ts
+   * export class MyController {
+   *   @get('/greet')
+   *   greet(@param.array('names', 'query', 'string') names: string[]): string {
+   *     return `Hello, ${names}`;
+   *   }
+   * }
+   * ```
+   *
+   * @param name Parameter name
+   * @param source Source of the parameter value
+   * @param itemSpec Item type for the array or the full item object
+   */
+  export const array = function(
+    name: string,
+    source: ParameterLocation,
+    itemSpec: SchemaObject | ReferenceObject,
```

Comment:
This should accept a string ('number') or a function (Number). Make sure to fix the jsdoc so that it reflects the change if you want to go through with it.

Reply:
^ the code is correct, jsdoc is out-of-date